<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="HiCExperiment">
<title>Introduction to HiCExperiment • HiCExperiment</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to HiCExperiment">
<meta property="og:description" content="HiCExperiment">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">HiCExperiment</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.11</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/HiCExperiment.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/js2264/HiCExperiment/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to HiCExperiment</h1>
                        <h4 data-toc-skip class="author">Jacques
Serizay</h4>
            
            <h4 data-toc-skip class="date">2023-03-09</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/js2264/HiCExperiment/blob/HEAD/vignettes/HiCExperiment.Rmd" class="external-link"><code>vignettes/HiCExperiment.Rmd</code></a></small>
      <div class="d-none name"><code>HiCExperiment.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Hi-C experimental approach allows one to query contact frequency for
all possible pairs of genomic loci simultaneously, in a genome-wide
manner. The output of this next-generation sequencing-supported
technique is a file describing every pair (a.k.a contact, or
interaction) between two genomic loci. This so-called “pairs” file can
be binned and transformed into a numerical matrix. In such matrix, each
cell contains the raw or normalized interaction
<strong>frequency</strong> between a pair of genomic loci (which
location can be retrieved using the corresponding column and row
indices).</p>
<p><a href="https://github.com/nservant/HiC-Pro" class="external-link">HiC-Pro</a>, <a href="https://github.com/open2c/distiller-nf" class="external-link">distiller</a> and <a href="https://github.com/aidenlab/juicer/" class="external-link">Juicer</a> are the three main
pipelines used to align, filter and process paired-end fastq reads into
pairs files and contact matrices. Each pipeline defined their own file
formats to store these two types of files.</p>
<ul>
<li><p>Pairs files are (gzipped) human-readable, text files that are a
variant of the BEDPE format; however the column order varies depending
on the pipeline being used.</p></li>
<li>
<p>Contact matrix file formats greatly vary depending on the
pipeline:</p>
<ul>
<li>
<code>HiC-Pro</code> generates two human-readable files: a
<code>regions</code> file describing each genomic interval, and a
<code>matrix</code> file quantifying interaction frequency between pairs
of loci from the <code>regions</code> file, using a standard triplet
sparse matrix format.</li>
<li>
<code>Juicer</code> generates a <code>.hic</code> file, a highly
compressed binary file storing sparse contact matrices from multiple
resolutions into a single file.</li>
<li>
<code>distiller</code> uses the <code>.(m)cool</code> format, a
sparse, compressed, binary genomic matrix data model built on HDF5.</li>
</ul>
</li>
</ul>
<p>Each file format can contain roughly the same information, albeit
with a largely improved compression for <code>.hic</code> and
<code>.(m)cool</code> files, which can also contain multi-resolution
matrices compared to the HiC-Pro derived files. The <a href="https://data.4dnucleome.org/help/about/about-dcic" class="external-link">4DN
consortium</a>, deciphering the role nuclear organization plays in gene
expression and cellular function, officially supports both the
<code>.hic</code> and <code>.(m)cool</code> formats. Furthermore, the
<code>.(m)cool</code> format has recently gained a lot of traction with
the release of a series of <code>python</code> packages
(<code>cooler</code>, <code>cooltools</code>, <code>pairtools</code>,
<code>coolpuppy</code>) by the <a href="https://open2c.github.io/" class="external-link">Open2C organization</a> facilitating
the investigation of Hi-C data stored in <code>.(m)cool</code> files in
a <code>python</code> environment.</p>
<p>The R <code>HiCExperiment</code> package aims at unlocking HiC
investigation within the rich, genomic-oriented Bioconductor
environment. It provides a set of classes and import functions to parse
HiC files (both contact matrices and pairs) in R, allowing random access
and efficient genome-based subsetting of contact matrices. It leverages
pre-existing base Bioconductor classes, notably
<code>GInteractions</code> and <code>ContactMatrix</code> classes (<a href="https://f1000research.com/articles/5-950/v2" class="external-link">Lun, Perry &amp;
Ing-Simmons, F1000 Research 2016</a>).</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><code>HiCExperiment</code> package can be installed from Bioconductor
using the following command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"HiCExperiment"</span><span class="op">)</span></span></code></pre></div>
<p>All R dependencies will be installed automatically.</p>
</div>
<div class="section level2">
<h2 id="the-hicexperiment-class">The <code>HiCExperiment</code> class<a class="anchor" aria-label="anchor" href="#the-hicexperiment-class"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiCExperiment" class="external-link">HiCExperiment</a></span><span class="op">)</span></span>
<span><span class="fu">showClass</span><span class="op">(</span><span class="st">"HiCExperiment"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Class "HiCExperiment" [package "HiCExperiment"]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Slots:</span></span>
<span><span class="co">#&gt;                                                                   </span></span>
<span><span class="co">#&gt; Name:             fileName               focus         resolutions</span></span>
<span><span class="co">#&gt; Class:           character     characterOrNULL             numeric</span></span>
<span><span class="co">#&gt;                                                                   </span></span>
<span><span class="co">#&gt; Name:           resolution        interactions              scores</span></span>
<span><span class="co">#&gt; Class:             numeric       GInteractions          SimpleList</span></span>
<span><span class="co">#&gt;                                                                   </span></span>
<span><span class="co">#&gt; Name:  topologicalFeatures           pairsFile            metadata</span></span>
<span><span class="co">#&gt; Class:          SimpleList     characterOrNULL                list</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Extends: "Annotated"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Known Subclasses: "AggrHiCExperiment"</span></span>
<span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/data.html">contacts_yeast</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">hic</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 8,757,906 contacts over 763 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b16f9fbb9a_7752" </span></span>
<span><span class="co">#&gt; focus: "whole genome" </span></span>
<span><span class="co">#&gt; resolutions(5): 1000 2000 4000 8000 16000</span></span>
<span><span class="co">#&gt; active resolution: 16000 </span></span>
<span><span class="co">#&gt; interactions: 267709 </span></span>
<span><span class="co">#&gt; scores(2): count balanced </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) centromeres(16) </span></span>
<span><span class="co">#&gt; pairsFile: N/A </span></span>
<span><span class="co">#&gt; metadata(0):</span></span></code></pre></div>
<p><img src="https://raw.githubusercontent.com/js2264/HiCExperiment/master/man/figures/HiCExperiment_data-structure.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="basics-importing--mcool--hic-or-hic-pro-generated-files-as-hicexperiment-objects">Basics: importing <code>.(m)cool</code>, <code>.hic</code> or
HiC-Pro-generated files as <code>HiCExperiment</code> objects<a class="anchor" aria-label="anchor" href="#basics-importing--mcool--hic-or-hic-pro-generated-files-as-hicexperiment-objects"></a>
</h2>
<div class="section level3">
<h3 id="import-methods">Import methods<a class="anchor" aria-label="anchor" href="#import-methods"></a>
</h3>
<p>The implemented <code><a href="../reference/import-methods.html">import()</a></code> methods allow one to import
Hi-C matrix files in R as <code>HiCExperiment</code> objects.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span></span>
<span>    <span class="st">"path/to/contact_matrix.cool"</span>, </span>
<span>    focus <span class="op">=</span> <span class="st">"chr:start-end"</span>, </span>
<span>    resolution <span class="op">=</span> <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>To give real-life examples, we use the <code>HiContactsData</code>
package to get access to a range of toy datasets available from the
<code>ExperimentHub</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/js2264/HiContactsData" class="external-link">HiContactsData</a></span><span class="op">)</span></span>
<span><span class="va">cool_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'cool'</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">cool_file</span>, format <span class="op">=</span> <span class="st">'cool'</span><span class="op">)</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 8,757,906 contacts over 12,079 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b13e94e7e8_7751" </span></span>
<span><span class="co">#&gt; focus: "whole genome" </span></span>
<span><span class="co">#&gt; resolutions(1): 1000</span></span>
<span><span class="co">#&gt; active resolution: 1000 </span></span>
<span><span class="co">#&gt; interactions: 2945692 </span></span>
<span><span class="co">#&gt; scores(2): count balanced </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) </span></span>
<span><span class="co">#&gt; pairsFile: N/A </span></span>
<span><span class="co">#&gt; metadata(0):</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="supporting-file-classes">Supporting file classes<a class="anchor" aria-label="anchor" href="#supporting-file-classes"></a>
</h3>
<p>There are currently three main standards to store Hi-C matrices in
files:</p>
<ul>
<li>
<code>.(m)cool</code> files</li>
<li>
<code>.hic</code> files</li>
<li>
<code>.matrix</code> and <code>.bed</code> files: generated by
HiC-Pro.</li>
</ul>
<p>Three supporting classes were specifically created to ensure that
each of these file structures would be properly parsed into
<code>HiCExperiment</code> objects:</p>
<ul>
<li><code>CoolFile</code></li>
<li><code>HicFile</code></li>
<li><code>HicproFile</code></li>
</ul>
<p>For each object, an optional <code>pairsFile</code> can be associated
and linked to the contact matrix file when imported as a
<code>HiCExperiment</code> object.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## --- CoolFile</span></span>
<span><span class="va">pairs_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'pairs.gz'</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">coolf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/CoolFile-class.html">CoolFile</a></span><span class="op">(</span><span class="va">cool_file</span>, pairsFile <span class="op">=</span> <span class="va">pairs_file</span><span class="op">)</span></span>
<span><span class="va">coolf</span></span>
<span><span class="co">#&gt; CoolFile object</span></span>
<span><span class="co">#&gt; .mcool file: /github/home/.cache/R/ExperimentHub/5b13e94e7e8_7751 </span></span>
<span><span class="co">#&gt; resolution: 1000 </span></span>
<span><span class="co">#&gt; pairs file: /github/home/.cache/R/ExperimentHub/5b15a2752d0_7753 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">coolf</span><span class="op">)</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 8,757,906 contacts over 12,079 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b13e94e7e8_7751" </span></span>
<span><span class="co">#&gt; focus: "whole genome" </span></span>
<span><span class="co">#&gt; resolutions(1): 1000</span></span>
<span><span class="co">#&gt; active resolution: 1000 </span></span>
<span><span class="co">#&gt; interactions: 2945692 </span></span>
<span><span class="co">#&gt; scores(2): count balanced </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) </span></span>
<span><span class="co">#&gt; pairsFile: /github/home/.cache/R/ExperimentHub/5b15a2752d0_7753 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="fu"><a href="../reference/AllGenerics.html">pairsFile</a></span><span class="op">(</span><span class="va">coolf</span><span class="op">)</span>, format <span class="op">=</span> <span class="st">'pairs'</span><span class="op">)</span></span>
<span><span class="co">#&gt; GInteractions object with 471364 interactions and 3 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames1   ranges1     seqnames2   ranges2 |     frag1     frag2</span></span>
<span><span class="co">#&gt;                &lt;Rle&gt; &lt;IRanges&gt;         &lt;Rle&gt; &lt;IRanges&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;        [1]        II       105 ---        II     48548 |      1358      1681</span></span>
<span><span class="co">#&gt;        [2]        II       113 ---        II     45003 |      1358      1658</span></span>
<span><span class="co">#&gt;        [3]        II       119 ---        II    687251 |      1358      5550</span></span>
<span><span class="co">#&gt;        [4]        II       160 ---        II     26124 |      1358      1510</span></span>
<span><span class="co">#&gt;        [5]        II       169 ---        II     39052 |      1358      1613</span></span>
<span><span class="co">#&gt;        ...       ...       ... ...       ...       ... .       ...       ...</span></span>
<span><span class="co">#&gt;   [471360]        II    808605 ---        II    809683 |      6316      6320</span></span>
<span><span class="co">#&gt;   [471361]        II    808609 ---        II    809917 |      6316      6324</span></span>
<span><span class="co">#&gt;   [471362]        II    808617 ---        II    809506 |      6316      6319</span></span>
<span><span class="co">#&gt;   [471363]        II    809447 ---        II    809685 |      6319      6321</span></span>
<span><span class="co">#&gt;   [471364]        II    809472 ---        II    809675 |      6319      6320</span></span>
<span><span class="co">#&gt;             distance</span></span>
<span><span class="co">#&gt;            &lt;integer&gt;</span></span>
<span><span class="co">#&gt;        [1]     48443</span></span>
<span><span class="co">#&gt;        [2]     44890</span></span>
<span><span class="co">#&gt;        [3]    687132</span></span>
<span><span class="co">#&gt;        [4]     25964</span></span>
<span><span class="co">#&gt;        [5]     38883</span></span>
<span><span class="co">#&gt;        ...       ...</span></span>
<span><span class="co">#&gt;   [471360]      1078</span></span>
<span><span class="co">#&gt;   [471361]      1308</span></span>
<span><span class="co">#&gt;   [471362]       889</span></span>
<span><span class="co">#&gt;   [471363]       238</span></span>
<span><span class="co">#&gt;   [471364]       203</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 549331 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="co">## --- HicFile</span></span>
<span><span class="va">hic_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'hic'</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">hicf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HicFile-class.html">HicFile</a></span><span class="op">(</span><span class="va">hic_file</span>, pairsFile <span class="op">=</span> <span class="va">pairs_file</span><span class="op">)</span></span>
<span><span class="va">hicf</span></span>
<span><span class="co">#&gt; HicFile object</span></span>
<span><span class="co">#&gt; .hic file: /github/home/.cache/R/ExperimentHub/5b17c7ffe79_7836 </span></span>
<span><span class="co">#&gt; resolution: 1000 </span></span>
<span><span class="co">#&gt; pairs file: /github/home/.cache/R/ExperimentHub/5b15a2752d0_7753 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">hicf</span><span class="op">)</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 13,681,280 contacts over 12,165 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b17c7ffe79_7836" </span></span>
<span><span class="co">#&gt; focus: "whole genome" </span></span>
<span><span class="co">#&gt; resolutions(5): 1000 2000 4000 8000 16000</span></span>
<span><span class="co">#&gt; active resolution: 1000 </span></span>
<span><span class="co">#&gt; interactions: 2965693 </span></span>
<span><span class="co">#&gt; scores(2): count balanced </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) </span></span>
<span><span class="co">#&gt; pairsFile: /github/home/.cache/R/ExperimentHub/5b15a2752d0_7753 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span></span>
<span><span class="co">## --- HicproFile</span></span>
<span><span class="va">hicpro_matrix_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'hicpro_matrix'</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">hicpro_regions_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'hicpro_bed'</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">hicprof</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/HicproFile-class.html">HicproFile</a></span><span class="op">(</span><span class="va">hicpro_matrix_file</span>, bed <span class="op">=</span> <span class="va">hicpro_regions_file</span><span class="op">)</span></span>
<span><span class="va">hicprof</span></span>
<span><span class="co">#&gt; HicproFile object</span></span>
<span><span class="co">#&gt; HiC-Pro files:</span></span>
<span><span class="co">#&gt;   $ matrix:   /github/home/.cache/R/ExperimentHub/5b121abc30d_7837 </span></span>
<span><span class="co">#&gt;   $ regions:  /github/home/.cache/R/ExperimentHub/5b17ecb70df_7838 </span></span>
<span><span class="co">#&gt; resolution: 1000 </span></span>
<span><span class="co">#&gt; pairs file: </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">hicprof</span><span class="op">)</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 9,503,604 contacts over 12,165 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b121abc30d_7837" </span></span>
<span><span class="co">#&gt; focus: "whole genome" </span></span>
<span><span class="co">#&gt; resolutions(1): 1000</span></span>
<span><span class="co">#&gt; active resolution: 1000 </span></span>
<span><span class="co">#&gt; interactions: 2686250 </span></span>
<span><span class="co">#&gt; scores(1): count </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) </span></span>
<span><span class="co">#&gt; pairsFile: N/A </span></span>
<span><span class="co">#&gt; metadata(1): regions</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="import-arguments">Import arguments<a class="anchor" aria-label="anchor" href="#import-arguments"></a>
</h2>
<div class="section level3">
<h3 id="querying-subsets-of-hi-c-matrix-files">Querying subsets of Hi-C matrix files<a class="anchor" aria-label="anchor" href="#querying-subsets-of-hi-c-matrix-files"></a>
</h3>
<p>The <code>focus</code> argument is used to specifically import
contacts within a genomic locus of interest.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/AllGenerics.html">availableChromosomes</a></span><span class="op">(</span><span class="va">cool_file</span><span class="op">)</span></span>
<span><span class="co">#&gt; Seqinfo object with 16 sequences from an unspecified genome:</span></span>
<span><span class="co">#&gt;   seqnames seqlengths isCircular genome</span></span>
<span><span class="co">#&gt;   I            230218       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   II           813184       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   III          316620       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   IV          1531933       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   V            576874       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ...             ...        ...    ...</span></span>
<span><span class="co">#&gt;   XII         1078177       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XIII         924431       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XIV          784333       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XV          1091291       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XVI          948066       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">cool_file</span>, format <span class="op">=</span> <span class="st">'cool'</span>,  focus <span class="op">=</span> <span class="st">'I:20001-80000'</span><span class="op">)</span></span>
<span><span class="va">hic</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 24,322 contacts over 60 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b13e94e7e8_7751" </span></span>
<span><span class="co">#&gt; focus: "I:20,001-80,000" </span></span>
<span><span class="co">#&gt; resolutions(1): 1000</span></span>
<span><span class="co">#&gt; active resolution: 1000 </span></span>
<span><span class="co">#&gt; interactions: 1653 </span></span>
<span><span class="co">#&gt; scores(2): count balanced </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) </span></span>
<span><span class="co">#&gt; pairsFile: N/A </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">focus</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "I:20001-80000"</span></span></code></pre></div>
<p><em>Note:</em><br>
Querying subsets of HiC-Pro formatted matrices is currently not
supported. HiC-Pro formatted matrices will systematically be fully
imported in memory when imported.</p>
<p>One can also extract a count matrix from a Hi-C matrix file that is
<em>not</em> centered at the diagonal. To do this, specify a couple of
coordinates in the <code>focus</code> argument using a character string
formatted as <code>"...|..."</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">cool_file</span>, format <span class="op">=</span> <span class="st">'cool'</span>, focus <span class="op">=</span> <span class="st">'II:1-500000|II:100001-300000'</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">focus</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "II:1-500000|II:100001-300000"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="multi-resolution-hi-c-matrix-files">Multi-resolution Hi-C matrix files<a class="anchor" aria-label="anchor" href="#multi-resolution-hi-c-matrix-files"></a>
</h3>
<p><code><a href="../reference/import-methods.html">import()</a></code> works with <code>.mcool</code> and
multi-resolution <code>.hic</code> files as well: in this case, the user
can specify the <code>resolution</code> at which count values are
recovered.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mcool_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/HiContactsData/man/HiContactsData.html" class="external-link">HiContactsData</a></span><span class="op">(</span><span class="st">'yeast_wt'</span>, format <span class="op">=</span> <span class="st">'mcool'</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2023-02-28</span></span>
<span><span class="co">#&gt; see ?HiContactsData and browseVignettes('HiContactsData') for documentation</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">availableResolutions</a></span><span class="op">(</span><span class="va">mcool_file</span><span class="op">)</span></span>
<span><span class="co">#&gt; resolutions(5): 1000 2000 4000 8000 16000</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">availableChromosomes</a></span><span class="op">(</span><span class="va">mcool_file</span><span class="op">)</span></span>
<span><span class="co">#&gt; Seqinfo object with 16 sequences from an unspecified genome:</span></span>
<span><span class="co">#&gt;   seqnames seqlengths isCircular genome</span></span>
<span><span class="co">#&gt;   I            230218       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   II           813184       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   III          316620       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   IV          1531933       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   V            576874       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ...             ...        ...    ...</span></span>
<span><span class="co">#&gt;   XII         1078177       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XIII         924431       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XIV          784333       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XV          1091291       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XVI          948066       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="va">hic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">mcool_file</span>, format <span class="op">=</span> <span class="st">'cool'</span>, focus <span class="op">=</span> <span class="st">'II:1-800000'</span>, resolution <span class="op">=</span> <span class="fl">2000</span><span class="op">)</span></span>
<span><span class="va">hic</span></span>
<span><span class="co">#&gt; `HiCExperiment` object with 466,123 contacts over 400 regions </span></span>
<span><span class="co">#&gt; -------</span></span>
<span><span class="co">#&gt; fileName: "/github/home/.cache/R/ExperimentHub/5b16f9fbb9a_7752" </span></span>
<span><span class="co">#&gt; focus: "II:1-800,000" </span></span>
<span><span class="co">#&gt; resolutions(5): 1000 2000 4000 8000 16000</span></span>
<span><span class="co">#&gt; active resolution: 2000 </span></span>
<span><span class="co">#&gt; interactions: 33479 </span></span>
<span><span class="co">#&gt; scores(2): count balanced </span></span>
<span><span class="co">#&gt; topologicalFeatures: compartments(0) borders(0) loops(0) viewpoints(0) </span></span>
<span><span class="co">#&gt; pairsFile: N/A </span></span>
<span><span class="co">#&gt; metadata(0):</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="hicexperiment-accessors">HiCExperiment accessors<a class="anchor" aria-label="anchor" href="#hicexperiment-accessors"></a>
</h2>
<div class="section level3">
<h3 id="slots">Slots<a class="anchor" aria-label="anchor" href="#slots"></a>
</h3>
<p>Slots for a <code>HiCExperiment</code> object can be accessed using
the following <code>getters</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">fileName</span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "/github/home/.cache/R/ExperimentHub/5b16f9fbb9a_7752"</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">focus</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "II:1-800000"</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">resolutions</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  1000  2000  4000  8000 16000</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">resolution</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 2000</span></span>
<span><span class="fu">interactions</span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; GInteractions object with 33479 interactions and 4 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames1       ranges1     seqnames2       ranges2 |   bin_id1</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;     &lt;IRanges&gt;         &lt;Rle&gt;     &lt;IRanges&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]        II        1-2000 ---        II        1-2000 |       116</span></span>
<span><span class="co">#&gt;       [2]        II        1-2000 ---        II     4001-6000 |       116</span></span>
<span><span class="co">#&gt;       [3]        II        1-2000 ---        II     6001-8000 |       116</span></span>
<span><span class="co">#&gt;       [4]        II        1-2000 ---        II    8001-10000 |       116</span></span>
<span><span class="co">#&gt;       [5]        II        1-2000 ---        II   10001-12000 |       116</span></span>
<span><span class="co">#&gt;       ...       ...           ... ...       ...           ... .       ...</span></span>
<span><span class="co">#&gt;   [33475]        II 794001-796000 ---        II 796001-798000 |       513</span></span>
<span><span class="co">#&gt;   [33476]        II 794001-796000 ---        II 798001-800000 |       513</span></span>
<span><span class="co">#&gt;   [33477]        II 796001-798000 ---        II 796001-798000 |       514</span></span>
<span><span class="co">#&gt;   [33478]        II 796001-798000 ---        II 798001-800000 |       514</span></span>
<span><span class="co">#&gt;   [33479]        II 798001-800000 ---        II 798001-800000 |       515</span></span>
<span><span class="co">#&gt;             bin_id2     count  balanced</span></span>
<span><span class="co">#&gt;           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]       116         1       NaN</span></span>
<span><span class="co">#&gt;       [2]       118         2       NaN</span></span>
<span><span class="co">#&gt;       [3]       119         3       NaN</span></span>
<span><span class="co">#&gt;       [4]       120        15       NaN</span></span>
<span><span class="co">#&gt;       [5]       121         9       NaN</span></span>
<span><span class="co">#&gt;       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [33475]       514       309 0.1194189</span></span>
<span><span class="co">#&gt;   [33476]       515       227 0.0956207</span></span>
<span><span class="co">#&gt;   [33477]       514       130 0.0501703</span></span>
<span><span class="co">#&gt;   [33478]       515       297 0.1249314</span></span>
<span><span class="co">#&gt;   [33479]       515       117 0.0536429</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 400 ranges and 4 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 16 sequences from an unspecified genome</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">scores</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 2</span></span>
<span><span class="co">#&gt; names(2): count balanced</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="../reference/AllGenerics.html">scores</a></span><span class="op">(</span><span class="va">hic</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 212 309 227 130 297 117</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="../reference/AllGenerics.html">scores</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">'balanced'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.08204677 0.11941893 0.09562069 0.05017035 0.12493137 0.05364290</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">topologicalFeatures</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 4</span></span>
<span><span class="co">#&gt; names(4): compartments borders loops viewpoints</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">pairsFile</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Annotated-class.html" class="external-link">metadata</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; list()</span></span></code></pre></div>
<p>Several extra functions are available as well:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">hic</span><span class="op">)</span> <span class="co">## To recover the `Seqinfo` object from the `.(m)cool` file</span></span>
<span><span class="co">#&gt; Seqinfo object with 16 sequences from an unspecified genome:</span></span>
<span><span class="co">#&gt;   seqnames seqlengths isCircular genome</span></span>
<span><span class="co">#&gt;   I            230218       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   II           813184       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   III          316620       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   IV          1531933       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   V            576874       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   ...             ...        ...    ...</span></span>
<span><span class="co">#&gt;   XII         1078177       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XIII         924431       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XIV          784333       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XV          1091291       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="co">#&gt;   XVI          948066       &lt;NA&gt;   &lt;NA&gt;</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">bins</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span> <span class="co">## To bin the genome at the current resolution</span></span>
<span><span class="co">#&gt; GRanges object with 6045 ranges and 2 metadata columns:</span></span>
<span><span class="co">#&gt;                     seqnames        ranges strand |    bin_id    weight</span></span>
<span><span class="co">#&gt;                        &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;            I_1_2000        I        1-2000      * |         0 0.0559613</span></span>
<span><span class="co">#&gt;         I_2001_4000        I     2001-4000      * |         1 0.0333136</span></span>
<span><span class="co">#&gt;         I_4001_6000        I     4001-6000      * |         2 0.0376028</span></span>
<span><span class="co">#&gt;         I_6001_8000        I     6001-8000      * |         3 0.0369553</span></span>
<span><span class="co">#&gt;        I_8001_10000        I    8001-10000      * |         4 0.0220139</span></span>
<span><span class="co">#&gt;                 ...      ...           ...    ... .       ...       ...</span></span>
<span><span class="co">#&gt;   XVI_940001_942000      XVI 940001-942000      * |      6040 0.0226033</span></span>
<span><span class="co">#&gt;   XVI_942001_944000      XVI 942001-944000      * |      6041       NaN</span></span>
<span><span class="co">#&gt;   XVI_944001_946000      XVI 944001-946000      * |      6042       NaN</span></span>
<span><span class="co">#&gt;   XVI_946001_948000      XVI 946001-948000      * |      6043       NaN</span></span>
<span><span class="co">#&gt;   XVI_948001_948066      XVI 948001-948066      * |      6044       NaN</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 16 sequences from an unspecified genome</span></span>
<span><span class="fu">regions</span><span class="op">(</span><span class="va">hic</span><span class="op">)</span> <span class="co">## To extract unique regions of the contact matrix</span></span>
<span><span class="co">#&gt; GRanges object with 400 ranges and 4 metadata columns:</span></span>
<span><span class="co">#&gt;                    seqnames        ranges strand |    bin_id    weight   chr</span></span>
<span><span class="co">#&gt;                       &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;          II_1_2000       II        1-2000      * |       116       NaN    II</span></span>
<span><span class="co">#&gt;       II_2001_4000       II     2001-4000      * |       117       NaN    II</span></span>
<span><span class="co">#&gt;       II_4001_6000       II     4001-6000      * |       118       NaN    II</span></span>
<span><span class="co">#&gt;       II_6001_8000       II     6001-8000      * |       119       NaN    II</span></span>
<span><span class="co">#&gt;      II_8001_10000       II    8001-10000      * |       120 0.0461112    II</span></span>
<span><span class="co">#&gt;                ...      ...           ...    ... .       ...       ...   ...</span></span>
<span><span class="co">#&gt;   II_790001_792000       II 790001-792000      * |       511 0.0236816    II</span></span>
<span><span class="co">#&gt;   II_792001_794000       II 792001-794000      * |       512 0.0272236    II</span></span>
<span><span class="co">#&gt;   II_794001_796000       II 794001-796000      * |       513 0.0196726    II</span></span>
<span><span class="co">#&gt;   II_796001_798000       II 796001-798000      * |       514 0.0196450    II</span></span>
<span><span class="co">#&gt;   II_798001_800000       II 798001-800000      * |       515 0.0214123    II</span></span>
<span><span class="co">#&gt;                       center</span></span>
<span><span class="co">#&gt;                    &lt;integer&gt;</span></span>
<span><span class="co">#&gt;          II_1_2000      1000</span></span>
<span><span class="co">#&gt;       II_2001_4000      3000</span></span>
<span><span class="co">#&gt;       II_4001_6000      5000</span></span>
<span><span class="co">#&gt;       II_6001_8000      7000</span></span>
<span><span class="co">#&gt;      II_8001_10000      9000</span></span>
<span><span class="co">#&gt;                ...       ...</span></span>
<span><span class="co">#&gt;   II_790001_792000    791000</span></span>
<span><span class="co">#&gt;   II_792001_794000    793000</span></span>
<span><span class="co">#&gt;   II_794001_796000    795000</span></span>
<span><span class="co">#&gt;   II_796001_798000    797000</span></span>
<span><span class="co">#&gt;   II_798001_800000    799000</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 16 sequences from an unspecified genome</span></span>
<span><span class="fu">anchors</span><span class="op">(</span><span class="va">hic</span><span class="op">)</span> <span class="co">## To extract "first" and "second" anchors for each interaction</span></span>
<span><span class="co">#&gt; $first</span></span>
<span><span class="co">#&gt; GRanges object with 33479 ranges and 4 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames        ranges strand |    bin_id    weight   chr    center</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;Rle&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;       [1]       II        1-2000      * |       116       NaN    II      1000</span></span>
<span><span class="co">#&gt;       [2]       II        1-2000      * |       116       NaN    II      1000</span></span>
<span><span class="co">#&gt;       [3]       II        1-2000      * |       116       NaN    II      1000</span></span>
<span><span class="co">#&gt;       [4]       II        1-2000      * |       116       NaN    II      1000</span></span>
<span><span class="co">#&gt;       [5]       II        1-2000      * |       116       NaN    II      1000</span></span>
<span><span class="co">#&gt;       ...      ...           ...    ... .       ...       ...   ...       ...</span></span>
<span><span class="co">#&gt;   [33475]       II 794001-796000      * |       513 0.0196726    II    795000</span></span>
<span><span class="co">#&gt;   [33476]       II 794001-796000      * |       513 0.0196726    II    795000</span></span>
<span><span class="co">#&gt;   [33477]       II 796001-798000      * |       514 0.0196450    II    797000</span></span>
<span><span class="co">#&gt;   [33478]       II 796001-798000      * |       514 0.0196450    II    797000</span></span>
<span><span class="co">#&gt;   [33479]       II 798001-800000      * |       515 0.0214123    II    799000</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 16 sequences from an unspecified genome</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $second</span></span>
<span><span class="co">#&gt; GRanges object with 33479 ranges and 4 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames        ranges strand |    bin_id    weight   chr    center</span></span>
<span><span class="co">#&gt;              &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;Rle&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;       [1]       II        1-2000      * |       116       NaN    II      1000</span></span>
<span><span class="co">#&gt;       [2]       II     4001-6000      * |       118       NaN    II      5000</span></span>
<span><span class="co">#&gt;       [3]       II     6001-8000      * |       119       NaN    II      7000</span></span>
<span><span class="co">#&gt;       [4]       II    8001-10000      * |       120 0.0461112    II      9000</span></span>
<span><span class="co">#&gt;       [5]       II   10001-12000      * |       121 0.0334807    II     11000</span></span>
<span><span class="co">#&gt;       ...      ...           ...    ... .       ...       ...   ...       ...</span></span>
<span><span class="co">#&gt;   [33475]       II 796001-798000      * |       514 0.0196450    II    797000</span></span>
<span><span class="co">#&gt;   [33476]       II 798001-800000      * |       515 0.0214123    II    799000</span></span>
<span><span class="co">#&gt;   [33477]       II 796001-798000      * |       514 0.0196450    II    797000</span></span>
<span><span class="co">#&gt;   [33478]       II 798001-800000      * |       515 0.0214123    II    799000</span></span>
<span><span class="co">#&gt;   [33479]       II 798001-800000      * |       515 0.0214123    II    799000</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 16 sequences from an unspecified genome</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="slot-setters">Slot setters<a class="anchor" aria-label="anchor" href="#slot-setters"></a>
</h3>
<div class="section level4">
<h4 id="scores">Scores<a class="anchor" aria-label="anchor" href="#scores"></a>
</h4>
<p>Add any <code>scores</code> metric using a numerical vector.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/AllGenerics.html">scores</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">'random'</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">scores</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 3</span></span>
<span><span class="co">#&gt; names(3): count balanced random</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">tail</a></span><span class="op">(</span><span class="fu"><a href="../reference/AllGenerics.html">scores</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">'random'</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.2839809 0.6531274 0.2934283 0.5204494 0.2204198 0.5712477</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h4>
<p>Add <code>topologicalFeatures</code> using <code>GRanges</code> or
<code>Pairs</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/AllGenerics.html">topologicalFeatures</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">'viewpoints'</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">GRanges</span><span class="op">(</span><span class="st">"II:300001-320000"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">topologicalFeatures</a></span><span class="op">(</span><span class="va">hic</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of length 4</span></span>
<span><span class="co">#&gt; names(4): compartments borders loops viewpoints</span></span>
<span><span class="fu"><a href="../reference/AllGenerics.html">topologicalFeatures</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">'viewpoints'</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 0 metadata columns:</span></span>
<span><span class="co">#&gt;       seqnames        ranges strand</span></span>
<span><span class="co">#&gt;          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   [1]       II 300001-320000      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="coercing-hicexperiment">Coercing <code>HiCExperiment</code><a class="anchor" aria-label="anchor" href="#coercing-hicexperiment"></a>
</h3>
<p>Using the <code><a href="../reference/as.html">as()</a></code> function, <code>HiCExperiment</code> can
be coerced in <code>GInteractions</code>, <code>ContactMatrix</code> and
<code>matrix</code> seamlessly.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/as.html">as</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">"GInteractions"</span><span class="op">)</span></span>
<span><span class="co">#&gt; GInteractions object with 33479 interactions and 5 metadata columns:</span></span>
<span><span class="co">#&gt;           seqnames1       ranges1     seqnames2       ranges2 |   bin_id1</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;     &lt;IRanges&gt;         &lt;Rle&gt;     &lt;IRanges&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]        II        1-2000 ---        II        1-2000 |       116</span></span>
<span><span class="co">#&gt;       [2]        II        1-2000 ---        II     4001-6000 |       116</span></span>
<span><span class="co">#&gt;       [3]        II        1-2000 ---        II     6001-8000 |       116</span></span>
<span><span class="co">#&gt;       [4]        II        1-2000 ---        II    8001-10000 |       116</span></span>
<span><span class="co">#&gt;       [5]        II        1-2000 ---        II   10001-12000 |       116</span></span>
<span><span class="co">#&gt;       ...       ...           ... ...       ...           ... .       ...</span></span>
<span><span class="co">#&gt;   [33475]        II 794001-796000 ---        II 796001-798000 |       513</span></span>
<span><span class="co">#&gt;   [33476]        II 794001-796000 ---        II 798001-800000 |       513</span></span>
<span><span class="co">#&gt;   [33477]        II 796001-798000 ---        II 796001-798000 |       514</span></span>
<span><span class="co">#&gt;   [33478]        II 796001-798000 ---        II 798001-800000 |       514</span></span>
<span><span class="co">#&gt;   [33479]        II 798001-800000 ---        II 798001-800000 |       515</span></span>
<span><span class="co">#&gt;             bin_id2     count  balanced    random</span></span>
<span><span class="co">#&gt;           &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;       [1]       116         1       NaN  0.463925</span></span>
<span><span class="co">#&gt;       [2]       118         2       NaN  0.517236</span></span>
<span><span class="co">#&gt;       [3]       119         3       NaN  0.433679</span></span>
<span><span class="co">#&gt;       [4]       120        15       NaN  0.516190</span></span>
<span><span class="co">#&gt;       [5]       121         9       NaN  0.832762</span></span>
<span><span class="co">#&gt;       ...       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt;   [33475]       514       309 0.1194189  0.653127</span></span>
<span><span class="co">#&gt;   [33476]       515       227 0.0956207  0.293428</span></span>
<span><span class="co">#&gt;   [33477]       514       130 0.0501703  0.520449</span></span>
<span><span class="co">#&gt;   [33478]       515       297 0.1249314  0.220420</span></span>
<span><span class="co">#&gt;   [33479]       515       117 0.0536429  0.571248</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 400 ranges and 4 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 16 sequences from an unspecified genome</span></span>
<span><span class="fu"><a href="../reference/as.html">as</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">"ContactMatrix"</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: ContactMatrix </span></span>
<span><span class="co">#&gt; dim: 400 400 </span></span>
<span><span class="co">#&gt; type: dgCMatrix </span></span>
<span><span class="co">#&gt; rownames: NULL</span></span>
<span><span class="co">#&gt; colnames: NULL</span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; regions: 400</span></span>
<span><span class="fu"><a href="../reference/as.html">as</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">"matrix"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span><span class="co">#&gt; 10 x 10 sparse Matrix of class "dgTMatrix"</span></span>
<span><span class="co">#&gt;                                                                           </span></span>
<span><span class="co">#&gt;  [1,] NaN . NaN NaN        NaN        NaN        NaN        NaN        NaN</span></span>
<span><span class="co">#&gt;  [2,]   . .   .   . .          .          .          .          .         </span></span>
<span><span class="co">#&gt;  [3,] NaN .   . NaN        NaN        NaN        NaN        NaN        NaN</span></span>
<span><span class="co">#&gt;  [4,] NaN . NaN NaN        NaN        NaN        NaN        NaN        NaN</span></span>
<span><span class="co">#&gt;  [5,] NaN . NaN NaN 0.08079721 0.18680431 0.13127403 0.08833001 0.06759757</span></span>
<span><span class="co">#&gt;  [6,] NaN . NaN NaN 0.18680431 0.08183011 0.19176749 0.12687633 0.10078115</span></span>
<span><span class="co">#&gt;  [7,] NaN . NaN NaN 0.13127403 0.19176749 0.08040523 0.13690173 0.13249106</span></span>
<span><span class="co">#&gt;  [8,] NaN . NaN NaN 0.08833001 0.12687633 0.13690173 0.07977117 0.18151495</span></span>
<span><span class="co">#&gt;  [9,] NaN . NaN NaN 0.06759757 0.10078115 0.13249106 0.18151495 0.06494950</span></span>
<span><span class="co">#&gt; [10,] NaN . NaN NaN 0.06021225 0.07728955 0.09404388 0.12720548 0.11622354</span></span>
<span><span class="co">#&gt;                 </span></span>
<span><span class="co">#&gt;  [1,]        NaN</span></span>
<span><span class="co">#&gt;  [2,] .         </span></span>
<span><span class="co">#&gt;  [3,]        NaN</span></span>
<span><span class="co">#&gt;  [4,]        NaN</span></span>
<span><span class="co">#&gt;  [5,] 0.06021225</span></span>
<span><span class="co">#&gt;  [6,] 0.07728955</span></span>
<span><span class="co">#&gt;  [7,] 0.09404388</span></span>
<span><span class="co">#&gt;  [8,] 0.12720548</span></span>
<span><span class="co">#&gt;  [9,] 0.11622354</span></span>
<span><span class="co">#&gt; [10,] 0.06796588</span></span>
<span><span class="fu"><a href="../reference/as.html">as</a></span><span class="op">(</span><span class="va">hic</span>, <span class="st">"data.frame"</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;    seqnames1 start1 end1 width1 strand1 bin_id1 weight1 center1 seqnames2</span></span>
<span><span class="co">#&gt; 1         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 2         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 3         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 4         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 5         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 6         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 7         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 8         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 9         II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt; 10        II      1 2000   2000       *     116     NaN    1000        II</span></span>
<span><span class="co">#&gt;    start2  end2 width2 strand2 bin_id2    weight2 center2 count balanced</span></span>
<span><span class="co">#&gt; 1       1  2000   2000       *     116        NaN    1000     1      NaN</span></span>
<span><span class="co">#&gt; 2    4001  6000   2000       *     118        NaN    5000     2      NaN</span></span>
<span><span class="co">#&gt; 3    6001  8000   2000       *     119        NaN    7000     3      NaN</span></span>
<span><span class="co">#&gt; 4    8001 10000   2000       *     120 0.04611120    9000    15      NaN</span></span>
<span><span class="co">#&gt; 5   10001 12000   2000       *     121 0.03348075   11000     9      NaN</span></span>
<span><span class="co">#&gt; 6   12001 14000   2000       *     122 0.03389168   13000     6      NaN</span></span>
<span><span class="co">#&gt; 7   14001 16000   2000       *     123 0.04164320   15000     1      NaN</span></span>
<span><span class="co">#&gt; 8   16001 18000   2000       *     124 0.01954625   17000     2      NaN</span></span>
<span><span class="co">#&gt; 9   18001 20000   2000       *     125 0.02331795   19000     6      NaN</span></span>
<span><span class="co">#&gt; 10  20001 22000   2000       *     126 0.02241734   21000     5      NaN</span></span>
<span><span class="co">#&gt;       random</span></span>
<span><span class="co">#&gt; 1  0.4639247</span></span>
<span><span class="co">#&gt; 2  0.5172362</span></span>
<span><span class="co">#&gt; 3  0.4336785</span></span>
<span><span class="co">#&gt; 4  0.5161904</span></span>
<span><span class="co">#&gt; 5  0.8327623</span></span>
<span><span class="co">#&gt; 6  0.4507897</span></span>
<span><span class="co">#&gt; 7  0.3646606</span></span>
<span><span class="co">#&gt; 8  0.8733799</span></span>
<span><span class="co">#&gt; 9  0.2883430</span></span>
<span><span class="co">#&gt; 10 0.2773951</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="importing-pairs-files">Importing pairs files<a class="anchor" aria-label="anchor" href="#importing-pairs-files"></a>
</h2>
<p>Pairs files typically contain chimeric pairs (filtered after
mapping), corresponding to loci that have been religated together after
restriction enzyme digestion. Such files have a variety of
standards.</p>
<ul>
<li>The <code>.pairs</code> file format, supported by the 4DN
consortium: <id><chr1><pos1><chr2><pos2><str1><str2> [<frag1><frag2>]</frag2></frag1></str2></str1></pos2></chr2></pos1></chr1></id>
</li>
<li>The pairs format generated by Juicer: [<readname>] <str1><chr1><pos1><frag1><str2><chr2><pos2><frag2> [<score>] [<mapq1><mapq2>]
[<mapq1><cigar1><sequence1><mapq2><cigar2><sequence2><readname1><readname2>]</readname2></readname1></sequence2></cigar2></mapq2></sequence1></cigar1></mapq1></mapq2></mapq1></score></frag2></pos2></chr2></str2></frag1></pos1></chr1></str1></readname>
</li>
<li>The <code>.(all)validPairs</code> file format, defined in the
HiC-Pro pipeline: <id><chr1><pos1><str1><chr2><pos2><str2><isize><frag1><frag2><mapq1><mapq2> [<allele-specific info>]</allele-specific></mapq2></mapq1></frag2></frag1></isize></str2></pos2></chr2></str1></pos1></chr1></id>
</li>
</ul>
<p>Pairs in any of these different formats are automatically detected
and imported in R with the <code>import</code> function:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/import-methods.html">import</a></span><span class="op">(</span><span class="va">pairs_file</span>, format <span class="op">=</span> <span class="st">'pairs'</span><span class="op">)</span></span>
<span><span class="co">#&gt; GInteractions object with 471364 interactions and 3 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames1   ranges1     seqnames2   ranges2 |     frag1     frag2</span></span>
<span><span class="co">#&gt;                &lt;Rle&gt; &lt;IRanges&gt;         &lt;Rle&gt; &lt;IRanges&gt; | &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;        [1]        II       105 ---        II     48548 |      1358      1681</span></span>
<span><span class="co">#&gt;        [2]        II       113 ---        II     45003 |      1358      1658</span></span>
<span><span class="co">#&gt;        [3]        II       119 ---        II    687251 |      1358      5550</span></span>
<span><span class="co">#&gt;        [4]        II       160 ---        II     26124 |      1358      1510</span></span>
<span><span class="co">#&gt;        [5]        II       169 ---        II     39052 |      1358      1613</span></span>
<span><span class="co">#&gt;        ...       ...       ... ...       ...       ... .       ...       ...</span></span>
<span><span class="co">#&gt;   [471360]        II    808605 ---        II    809683 |      6316      6320</span></span>
<span><span class="co">#&gt;   [471361]        II    808609 ---        II    809917 |      6316      6324</span></span>
<span><span class="co">#&gt;   [471362]        II    808617 ---        II    809506 |      6316      6319</span></span>
<span><span class="co">#&gt;   [471363]        II    809447 ---        II    809685 |      6319      6321</span></span>
<span><span class="co">#&gt;   [471364]        II    809472 ---        II    809675 |      6319      6320</span></span>
<span><span class="co">#&gt;             distance</span></span>
<span><span class="co">#&gt;            &lt;integer&gt;</span></span>
<span><span class="co">#&gt;        [1]     48443</span></span>
<span><span class="co">#&gt;        [2]     44890</span></span>
<span><span class="co">#&gt;        [3]    687132</span></span>
<span><span class="co">#&gt;        [4]     25964</span></span>
<span><span class="co">#&gt;        [5]     38883</span></span>
<span><span class="co">#&gt;        ...       ...</span></span>
<span><span class="co">#&gt;   [471360]      1078</span></span>
<span><span class="co">#&gt;   [471361]      1308</span></span>
<span><span class="co">#&gt;   [471362]       889</span></span>
<span><span class="co">#&gt;   [471363]       238</span></span>
<span><span class="co">#&gt;   [471364]       203</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   regions: 549331 ranges and 0 metadata columns</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="further-documentation">Further documentation<a class="anchor" aria-label="anchor" href="#further-documentation"></a>
</h2>
<p>Please check <code><a href="../reference/HiCExperiment.html">?HiCExperiment</a></code> in R for a full description
of available slots, getters and setters, and comprehensive examples of
interaction with a HiCExperiment object.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R Under development (unstable) (2023-03-02 r83926)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] HiCExperiment_0.99.11 HiContactsData_1.1.9  ExperimentHub_2.7.1  </span></span>
<span><span class="co">#&gt;  [4] AnnotationHub_3.7.3   BiocFileCache_2.7.2   dbplyr_2.3.1         </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.51.4  GenomeInfoDb_1.35.15  IRanges_2.33.0       </span></span>
<span><span class="co">#&gt; [10] S4Vectors_0.37.4      BiocGenerics_0.45.0   dplyr_1.1.0          </span></span>
<span><span class="co">#&gt; [13] BiocStyle_2.27.1     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] tidyselect_1.2.0              blob_1.2.3                   </span></span>
<span><span class="co">#&gt;  [3] filelock_1.0.2                Biostrings_2.67.0            </span></span>
<span><span class="co">#&gt;  [5] bitops_1.0-7                  fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [7] RCurl_1.98-1.10               promises_1.2.0.1             </span></span>
<span><span class="co">#&gt;  [9] digest_0.6.31                 mime_0.12                    </span></span>
<span><span class="co">#&gt; [11] lifecycle_1.0.3               ellipsis_0.3.2               </span></span>
<span><span class="co">#&gt; [13] KEGGREST_1.39.0               interactiveDisplayBase_1.37.0</span></span>
<span><span class="co">#&gt; [15] RSQLite_2.3.0                 magrittr_2.0.3               </span></span>
<span><span class="co">#&gt; [17] compiler_4.3.0                rlang_1.0.6                  </span></span>
<span><span class="co">#&gt; [19] sass_0.4.5                    tools_4.3.0                  </span></span>
<span><span class="co">#&gt; [21] utf8_1.2.3                    yaml_2.3.7                   </span></span>
<span><span class="co">#&gt; [23] knitr_1.42                    bit_4.0.5                    </span></span>
<span><span class="co">#&gt; [25] curl_5.0.0                    DelayedArray_0.25.0          </span></span>
<span><span class="co">#&gt; [27] BiocParallel_1.33.9           withr_2.5.0                  </span></span>
<span><span class="co">#&gt; [29] purrr_1.0.1                   desc_1.4.2                   </span></span>
<span><span class="co">#&gt; [31] grid_4.3.0                    fansi_1.0.4                  </span></span>
<span><span class="co">#&gt; [33] xtable_1.8-4                  Rhdf5lib_1.21.0              </span></span>
<span><span class="co">#&gt; [35] SummarizedExperiment_1.29.1   cli_3.6.0                    </span></span>
<span><span class="co">#&gt; [37] rmarkdown_2.20                crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [39] ragg_1.2.5                    generics_0.1.3               </span></span>
<span><span class="co">#&gt; [41] tzdb_0.3.0                    httr_1.4.5                   </span></span>
<span><span class="co">#&gt; [43] rhdf5_2.43.1                  DBI_1.1.3                    </span></span>
<span><span class="co">#&gt; [45] cachem_1.0.7                  stringr_1.5.0                </span></span>
<span><span class="co">#&gt; [47] zlibbioc_1.45.0               parallel_4.3.0               </span></span>
<span><span class="co">#&gt; [49] AnnotationDbi_1.61.0          BiocManager_1.30.20          </span></span>
<span><span class="co">#&gt; [51] XVector_0.39.0                matrixStats_0.63.0           </span></span>
<span><span class="co">#&gt; [53] vctrs_0.5.2                   Matrix_1.5-3                 </span></span>
<span><span class="co">#&gt; [55] jsonlite_1.8.4                bookdown_0.33                </span></span>
<span><span class="co">#&gt; [57] bit64_4.0.5                   systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [59] strawr_0.0.9                  jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt; [61] glue_1.6.2                    pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt; [63] codetools_0.2-19              stringi_1.7.12               </span></span>
<span><span class="co">#&gt; [65] BiocVersion_3.17.1            later_1.3.0                  </span></span>
<span><span class="co">#&gt; [67] BiocIO_1.9.2                  tibble_3.2.0                 </span></span>
<span><span class="co">#&gt; [69] pillar_1.8.1                  rhdf5filters_1.11.1          </span></span>
<span><span class="co">#&gt; [71] rappdirs_0.3.3                htmltools_0.5.4              </span></span>
<span><span class="co">#&gt; [73] GenomeInfoDbData_1.2.9        R6_2.5.1                     </span></span>
<span><span class="co">#&gt; [75] textshaping_0.3.6             rprojroot_2.0.3              </span></span>
<span><span class="co">#&gt; [77] vroom_1.6.1                   lattice_0.20-45              </span></span>
<span><span class="co">#&gt; [79] evaluate_0.20                 shiny_1.7.4                  </span></span>
<span><span class="co">#&gt; [81] Biobase_2.59.0                highr_0.10                   </span></span>
<span><span class="co">#&gt; [83] png_0.1-8                     memoise_2.0.1                </span></span>
<span><span class="co">#&gt; [85] httpuv_1.6.9                  bslib_0.4.2                  </span></span>
<span><span class="co">#&gt; [87] Rcpp_1.0.10                   InteractionSet_1.27.0        </span></span>
<span><span class="co">#&gt; [89] xfun_0.37                     fs_1.6.1                     </span></span>
<span><span class="co">#&gt; [91] MatrixGenerics_1.11.0         pkgconfig_2.0.3</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jacques Serizay.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
